tests/src/JIT/HardwareIntrinsics/X86/Shared/SimpleBinOpTest.template

```diff
@@ -0,0 +1,216 @@
+        }}
+
+        public void RunReflectionScenario()
+        {{
+            var result = typeof({0}).GetMethod(nameof({0}.{1}), new Type[] {{ typeof({3}<{2}>), typeof({3}<{2}>) }})
+                                     .Invoke(null, new object[] {{
+                                        Unsafe.Read<{3}<{2}>>(_dataTable.inArray1Ptr),
+                                        Unsafe.Read<{3}<{2}>>(_dataTable.inArray2Ptr)
+                                     }});
+
+            Unsafe.Write(_dataTable.outArrayPtr, ({3}<{2}>)(result));
+            ValidateResult(_dataTable.inArray1, _dataTable.inArray2, _dataTable.outArray);
+        }}
+
+        public void RunClsVarScenario()
+        {{
+            var result = {0}.{1}(
+                _clsVar1,
+                _clsVar2
+            );
+
+            Unsafe.Write(_dataTable.outArrayPtr, result);
+            ValidateResult(_clsVar1, _clsVar2, _dataTable.outArray);
+        }}
+
+        public void RunLclVarScenario()
+        {{
+            var left = Unsafe.Read<{3}<{2}>>(_dataTable.inArray1Ptr);
+            var right = Unsafe.Read<{3}<{2}>>(_dataTable.inArray2Ptr);
+            var result = {0}.{1}(left, right);
+
+            Unsafe.Write(_dataTable.outArrayPtr, result);
+            ValidateResult(left, right, _dataTable.outArray);
+        }}
+
+        public void RunLclFldScenario()
+        {{
+            var test = new SimpleBinaryOpTest__{1}{2}();
+            var result = {0}.{1}(test._fld1, test._fld2);
+
+            Unsafe.Write(_dataTable.outArrayPtr, result);
+            ValidateResult(test._fld1, test._fld2, _dataTable.outArray);
+        }}
+
+        public void RunFldScenario()
+        {{
+            var result = {0}.{1}(_fld1, _fld2);
+
+            Unsafe.Write(_dataTable.outArrayPtr, result);
+            ValidateResult(_fld1, _fld2, _dataTable.outArray);
+        }}
+
+        public void RunUnsupportedScenario()
+        {{
+            Succeeded = false;
+
+            try
+            {{
+                RunBasicScenario();
+            }}
+            catch (PlatformNotSupportedException)
+            {{
+                Succeeded = true;
+            }}
+        }}
+
+        private void ValidateResult({3}<{2}> left, {3}<{2}> right, {2}[] result, [CallerMemberName] string method = "")
+        {{
+            {2}[] inArray1 = new {2}[ElementCount];
+            {2}[] inArray2 = new {2}[ElementCount];
+
+            Unsafe.Write(Unsafe.AsPointer(ref inArray1[0]), left);
+            Unsafe.Write(Unsafe.AsPointer(ref inArray2[0]), right);
+
+            ValidateResult(inArray1, inArray2, result, method);
+        }}
+
+        private void ValidateResult({2}[] left, {2}[] right, {2}[] result, [CallerMemberName] string method = "")
+        {{
+            if ({6})
+            {{
+                Succeeded = false;
+            }}
+            else
+            {{
+                for (var i = 1; i < left.Length; i++)
```

Comment:
I think we can remove this `for` loop to enable the template for more intrinsics (e.g., `HorizontalAdd`).

Reply:
You mean just check each element individually, rather than looping?

Reply:
Yes, that would make the template more general.

Reply:
I'm going to log an issue to track this (as part of porting the remaining tests to use the template as well).

We probably want to refactor it such that we don't have to duplicate a bunch of table data for the regular intrinsics (where each check will be the same).